services/sts/src/main/java/software/amazon/awssdk/services/sts/internal/StsAuthUtils.java

```diff
@@ -31,9 +31,11 @@ public static String accountIdFromArn(AssumedRoleUser assumedRoleUser) {
         if (assumedRoleUser == null) {
             return null;
         }
-        return Arn.parse(assumedRoleUser.arn())
-                  .map(Arn::accountId)
-                  .orElse(null);
+        RuleArn arn = RuleArn.parse(assumedRoleUser.arn());
+        if (arn == null) {
```

Comment:
nit : would be great to extract this into a method arnAccountId(RuleArn arn);

```
RuleArn arn = RuleArn.parse(assumedRoleUser.arn()); //or
RuleArn arn = RuleArn.parse(federatedUser.arn());

return arnAccountId(RuleArn arn);
```

Reply:
Since they're in different classes (so the code is harder to share) and there's only 2 places it's done like this (so [the rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) isn't satisfied), I'd usually leave it like this. I don't feel strongly about it, though, so LMK if you'd prefer it to be shared. I'd probably do a convenience method on RuleArn, like RuleArn.accountId(arn).